tensorflow_addons/layers/normalizations.py

```diff
@@ -321,3 +321,214 @@ def __init__(self, **kwargs):
+class FilterResponseNormalization(tf.keras.layers.Layer):
+    """Filter response normalization layer.
+
+    Filter Response Normalization (FRN), a normalization
+    method that enables models trained with per-channel
+    normalization to achieve high accuracy. It performs better than
+    all other normalization techniques for small batches and is par
+    with Batch Normalization for bigger batch sizes.
+
+    Arguments
+        axis: List of axes that should be normalized. This should represent the
+              spatial dimensions.
+        epsilon: Small float value added to variance to avoid dividing by zero.
+        beta_initializer: Initializer for the beta weight.
+        gamma_initializer: Initializer for the gamma weight.
+        beta_regularizer: Optional regularizer for the beta weight.
+        gamma_regularizer: Optional regularizer for the gamma weight.
+        beta_constraint: Optional constraint for the beta weight.
+        gamma_constraint: Optional constraint for the gamma weight.
+        learned_epsilon: (bool) Whether to add another learnable
+        epsilon parameter or not.
+        name: Optional name for the layer
+
+    Input shape
+        Arbitrary. Use the keyword argument `input_shape`
+        (tuple of integers, does not include the samples axis)
+        when using this layer as the first layer in a model. This layer, as of now,
+        works on a 4-D tensor where the tensor should have the shape [N X H X W X C]
+
+        TODO: Add support for more input shapes, especially for FC layers.
+
+    Output shape
+        Same shape as input.
+
+    References
+        - [Filter Response Normalization Layer: Eliminating Batch Dependence
+        in the training of Deep Neural Networks]
+        (https://arxiv.org/abs/1911.09737)
+    """
+
+    def __init__(
+        self,
+        epsilon: float = 1e-6,
+        axis: list = [1, 2],
+        beta_initializer: types.Initializer = "zeros",
+        gamma_initializer: types.Initializer = "ones",
+        beta_regularizer: types.Regularizer = None,
+        gamma_regularizer: types.Regularizer = None,
+        beta_constraint: types.Constraint = None,
+        gamma_constraint: types.Constraint = None,
+        learned_epsilon: bool = False,
+        name: str = None,
+        **kwargs
+    ):
+        super().__init__(name=name, **kwargs)
+        self.epsilon = tf.cast(epsilon, dtype=self.dtype)
+        self.beta_initializer = tf.keras.initializers.get(beta_initializer)
+        self.gamma_initializer = tf.keras.initializers.get(gamma_initializer)
+        self.beta_regularizer = tf.keras.regularizers.get(beta_regularizer)
+        self.gamma_regularizer = tf.keras.regularizers.get(gamma_regularizer)
+        self.beta_constraint = tf.keras.constraints.get(beta_constraint)
+        self.gamma_constraint = tf.keras.constraints.get(gamma_constraint)
+        self.use_eps_learned = learned_epsilon
+        self.supports_masking = True
```

Comment:
If this layer supports masking, maybe we should have a simple test for that?

Reply:
Ok

Reply:
Right now the layer supports only 4D(CNNs) only, so testing masking isn't meaningful right now but in the future we are planning to add support for more shapes. Once that is done, we can add test for it then

Reply:
Then I propose that we set `self.supports_masking = False` until we supports shapes where it's meaningful to have a mask. What do you think?

Reply:
I can do that for sure.